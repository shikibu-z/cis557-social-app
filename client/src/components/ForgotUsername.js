import {
  Modal, Form,
} from 'react-bootstrap';
import React, { useState } from 'react';

function ForgotUsername(props) {
  const [email, setEmail] = useState(null);
  const [sent, setSent] = useState(false);
  const onSend = () => {
    // handle sending email
    setSent(true);
  };
  const { show, onHide } = props;
  const onClose = () => {
    setSent(false);
    onHide();
  };
  return (
    <Modal show={show} onHide={onClose}>
      <Modal.Header closeButton />
      <Modal.Body className="ModalDialog">
        {!sent ? (
          <Form>
            <Form.Group>
              <Form.Label>Send Username to this email</Form.Label>
              <Form.Control type="email" className="form-input" value={email} onChange={(e) => setEmail(e.target.value)} />
            </Form.Group>
            <div className="button-group">
              <button className="btn" type="button" onClick={onSend}>
                Send
              </button>
            </div>
          </Form>
        ) : (
          <h2 style={{ 'text-align': 'center', width: '280px', height: '140px' }}>Email Sent</h2>
        )}

      </Modal.Body>
    </Modal>
  );
}

export default ForgotUsername;
