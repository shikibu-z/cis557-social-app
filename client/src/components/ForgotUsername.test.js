/*eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import ForgotUsername from './ForgotUsername';

test('ForgotUsername popup matches snapshot', () => {
  const component = renderer.create(<ForgotUsername />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
