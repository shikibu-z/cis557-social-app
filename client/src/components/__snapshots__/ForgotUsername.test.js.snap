// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotUsername popup matches snapshot 1`] = `null`;
